File: app/src/main/java/me/devsaki/hentoid/activities/sources/MultpornActivity.java
package me.devsaki.hentoid.activities.sources;

import me.devsaki.hentoid.enums.Site;

public class MultpornActivity extends BaseWebActivity {

    private static final String DOMAIN_FILTER = "multporn.net";
    private static final String[] GALLERY_FILTER = {"multporn.net/node/[0-9]+$", "multporn.net/(hentai_manga|hentai|comics|pictures|rule_6|gay_porn_comics|GIF)/[\\w%_\\-]+$"};
    private static final String[] JS_WHITELIST = {DOMAIN_FILTER};
    private static final String[] JS_CONTENT_BLACKLIST = {"exoloader", "popunder"};
    private static final String[] AD_ELEMENTS = {"iframe", ".c-ads"};


    Site getStartSite() {
        return Site.MULTPORN;
    }

    @Override
    protected CustomWebViewClient createWebClient() {
        CustomWebViewClient client = new CustomWebViewClient(getStartSite(), GALLERY_FILTER, this);
        client.restrictTo(DOMAIN_FILTER);
        client.addRemovableElements(AD_ELEMENTS);
        client.addJavascriptBlacklist(JS_CONTENT_BLACKLIST);
        client.adBlocker.addToJsUrlWhitelist(JS_WHITELIST);
        //for (String s : JS_CONTENT_BLACKLIST) client.adBlocker.addJsContentBlacklist(s);
        return client;
    }
}

File: app/src/main/java/me/devsaki/hentoid/activities/sources/MultpornActivity.kt
package me.devsaki.hentoid.activities.sources

import me.devsaki.hentoid.enums.Site

class MultpornActivity : BaseWebActivity() {

    companion object {
        private const val DOMAIN_FILTER = "multporn.net"
        private val GALLERY_FILTER = arrayOf(
            "multporn.net/node/[0-9]+$",
            "multporn.net/(hentai_manga|authors|hentai|comics|pictures|rule_6|gay_porn_comics|GIF)/[\\w%_\\-]+$",
            "multporn.net/node/[0-9]+$",
            "multporn.net/(hentai_manga|authors|hentai|comics|pictures|rule_6|gay_porn_comics|GIF)/[\\w%_\\-]+/[\\w%_\\-]+$"
        )
        private val JS_WHITELIST = arrayOf(DOMAIN_FILTER)
        private val JS_CONTENT_BLACKLIST = arrayOf("exoloader", "popunder")
        private val AD_ELEMENTS = arrayOf("iframe", ".c-ads")
    }


    override fun getStartSite(): Site {
        return Site.MULTPORN
    }

    override fun createWebClient(): CustomWebViewClient {
        val client = CustomWebViewClient(getStartSite(), GALLERY_FILTER, this)
        client.restrictTo(DOMAIN_FILTER)
        client.addRemovableElements(*AD_ELEMENTS)
        client.addJavascriptBlacklist(*JS_CONTENT_BLACKLIST)
        client.adBlocker.addToJsUrlWhitelist(*JS_WHITELIST)
        //for (String s : JS_CONTENT_BLACKLIST) client.adBlocker.addJsContentBlacklist(s);
        return client
    }
}